tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/NetStandardLibraries/ApiCalls.cs

```diff
@@ -62,5 +63,68 @@ public static void TestSetTransactionUserId(string userId)
         {
             NewRelic.Api.Agent.NewRelic.GetAgent().CurrentTransaction.SetUserId(userId);
         }
+
+        [LibraryMethod]
+        [MethodImpl(MethodImplOptions.NoOptimization | MethodImplOptions.NoInlining)]
+        public static void TestSetErrorGroupCallbackReturnsString(string callbackReturn)
+        {
+            NewRelic.Api.Agent.NewRelic.SetErrorGroupCallback((x) => callbackReturn);
+        }
+
+        [LibraryMethod]
+        [MethodImpl(MethodImplOptions.NoOptimization | MethodImplOptions.NoInlining)]
+        public static void TestSetErrorGroupCallbackWithKeys()
+        {
+            // list of keys that are expected as part of agent spec
+            var expectedKeys = new[]
+            {
+                "error.class",
+                "error.message",
+                "stack_trace",
+                "transactionName",
+                "request.uri",
+                "error.expected"
+                // The following attribute keys are expected to be available, but in this specific test, they are not.
+                // "http.statusCode", (no http call in test)
+                // "request.method" (no http call in test)
+                // "transactionUiName" (Not available)
+            };
+
+            NewRelic.Api.Agent.NewRelic.SetErrorGroupCallback((dict) =>
+            {
+                if (dict == null)
+                {
+                    return "IReadOnlyDictionary was null";
+                }
+
+                if (dict.Count < 1)
+                {
+                    return "IReadOnlyDictionary Count was " + dict.Count;
+                }
+
+                var missingKeys = new List<string>();
+                foreach (var key in expectedKeys)
+                {
+                    if (!dict.ContainsKey(key))
+                    {
+                        missingKeys.Add(key);
+                    }
+                }
+
+                if (missingKeys.Count > 0)
```

Comment:
Do you want to also check that there aren't any unexpected extra keys?

Reply:
The number of keys is not fixed.  We want as many as we can get, but at least the once mentioned in the spec.